Core/Src/main.c

```diff
@@ -87,17 +98,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // SPI
+  SPI_HandleTypeDef *pSPI_Handle = &hspi1;
+  uint8_t *TxBuffer = malloc(BYTE_COUNT * sizeof(uint8_t));
+  uint8_t *RxBuffer = malloc(BYTE_COUNT * sizeof(uint8_t));
```

Comment:
any reason why you are using malloc here? In general within embedded systems we try to avoid using dynamic memory allocations as it COULD lead to a lot of issues. In this case, statically allocating an array would do just fine, but this implementation would still work.